docs/source/user_guide/best_practices.rst

```diff
@@ -0,0 +1,167 @@
+==============
+Best Practices
+==============
+
+Leverage PySpark APIs
+---------------------
+
+Koalas uses Spark under the hood; therefore, many features and performance optimization are available
+in Koalas as well. Leverage and combine those cutting-edge features with Koalas.
+
+Existing Spark context and Spark sessions are used out of the box in Koalas. If you already have your own
+configured Spark context or sessions running, Koalas uses them.
+
+If there is no Spark context or session running in your environment (e.g., ordinary Python interpretor),
+such configurations can be set to ``SparkContext`` and/or ``SparkSession``.
+Once Spark context and/or session is created, Koalas can use this context and/or session automatically.
+For example, if you want to configure the executor memory in Spark, you can do as below:
+
+.. code-block:: python
+
+   from pyspark import SparkConf, SparkContext
+   conf = SparkConf()
+   conf.set('spark.executor.memory', '2g')
+   # Koalas automatically uses this Spark context with the configurations set.
+   SparkContext(conf=conf)
+
+   import databricks.koalas as ks
+   ...
+
+Another common configuration might be Arrow optimization in PySpark. In case of SQL configuration,
+it can be set into Spark session as below:
+
+.. code-block:: python
+
+   from pyspark.sql import SparkSession
+   builder = SparkSession.builder.appName("Koalas")
+   builder = builder.config("spark.sql.execution.arrow.enabled", "true")
+   # Koalas automatically uses this Spark session with the configurations set.
+   builder.getOrCreate()
+
+   import databricks.koalas as ks
+   ...
+
+All Spark features such as history server, web UI and deployment modes can be used as are with Koalas.
+If you are interested in performance turning, please see also `Tuning Spark <https://spark.apache.org/docs/latest/tuning.html>`_.
+
+
+Check execution plans
+---------------------
+
+Expensive operations can be predicted by leveraging PySpark API `to_spark().explain()`
+before the actual computation since Koalas is based on lazy execution. For example, see below.
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> kdf = ks.DataFrame({'id': range(10)})
+   >>> kdf = kdf[kdf.id > 5]
+   >>> kdf.to_spark().explain()
```

Comment:
As we saw at #1033, `to_spark().explain()` looses index columns and the result might be significantly different with default index due to column pruning. E.g.,

```py
>>> kdf = ks.range(10)
>>> kdf = kdf[kdf.id > 5]

>>> kdf.to_spark().explain()
== Physical Plan ==
*(1) Filter (id#28L > 5)
+- *(1) Range (0, 10, step=1, splits=12)

>>> kdf.to_spark('index').explain()
== Physical Plan ==
*(3) Project [(_we0#33 - 1) AS index#41, id#28L]
+- *(3) Filter (id#28L > 5)
   +- Window [row_number() windowspecdefinition(_w0#32L ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS _we0#33], [_w0#32L ASC NULLS FIRST]
      +- *(2) Sort [_w0#32L ASC NULLS FIRST], false, 0
         +- Exchange SinglePartition
            +- *(1) Project [id#28L, monotonically_increasing_id() AS _w0#32L]
               +- *(1) Range (0, 10, step=1, splits=12)
```

Reply:
We might want to have our own `explain()` to include index columns automatically?

Reply:
I submitted a PR #1042.

Reply:
Yeah, I think it's a good point. Let me take a look for that.